database/raft/tls.go

```diff
@@ -0,0 +1,29 @@
+package raft
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+
+	"chain/errors"
+)
+
+func verifyTLSName(name string, client *http.Client) error {
+	c := clientTLS(client)
+	if c == nil {
+		return nil
+	}
+	x509Cert, err := x509.ParseCertificate(c.Certificates[0].Certificate[0])
```

Comment:
If there's a TLS client config at all here, it should definitely have a cert, so I think crashing is fine.